build.gradle

```diff
@@ -71,18 +70,16 @@ task configureGroovyAll {
             'MANIFEST_MF'                   : 'META-INF/MANIFEST.MF'
         ]
 
-        def collectDuplicates = { File[] roots ->
+        def findDuplicates = { List<FileTree> fileForest ->
             Map<String, List<File>> duplicates = [:].withDefault { [] }
-            roots.each { root ->
-                root.eachFileRecurse(FileType.FILES) {
-                    def relative = root.toURI().relativize(it.toURI())
-                    duplicates[relative.toString()] += it
-                }
-            }
-            return duplicates.findAll { it.value.size() > 1 }
+            fileForest*.visit([
+                visitDir : {},
+                visitFile: { duplicates[it.relativePath.toString()] << it.file }
+            ] as FileVisitor)
+            duplicates.findAll { it.value.size() > 1 }
```

Comment:
The local variable should be called something like `files` or `filesByPath` or something, as it doesn't always contain duplicates.

Reply:
Renamed to `filesByRelativePath`